src/app/workflow/executions/executions-tab.component.ts

```diff
@@ -0,0 +1,210 @@
+        },
+        (error) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  private setMetricsObject(metricsMap: Map<PartnerEnum, Metrics>, partner: string, metric: Metrics) {
+    for (const partnerEnum in PartnerEnum) {
+      if (PartnerEnum[partnerEnum] === partner) {
+        metricsMap.set(PartnerEnum[partnerEnum], metric);
+      }
+    }
+  }
+
+  private resetMetricsData() {
+    this.metricsExist = false;
+    this.metrics = new Map();
+    this.metricsTable = [];
+    this.validationsTable = [];
+    this.partners = [];
+    this.totalExecutions = null;
+    this.successfulExecutions = null;
+    this.failedExecutions = null;
+    this.currentPartner = null;
+    this.metricsColumns = null;
+    this.validationsColumns = null;
+  }
+
+  private loadMetricsData(partner: PartnerEnum) {
+    this.metricsTable = [];
+    const metrics = this.metrics.get(partner);
+    if (metrics) {
+      this.metricsColumns = ['metric', 'min', 'avg', 'max'];
+      this.insertToMetricsTable(this.metricsTable, metrics?.cpu, 'CPU');
+      this.insertToMetricsTable(this.metricsTable, metrics?.memory, 'Memory');
+      this.insertToMetricsTable(this.metricsTable, metrics?.executionTime, 'Run Time');
+      if (metrics.executionStatusCount) {
+        this.totalExecutions =
+          metrics.executionStatusCount.numberOfSuccessfulExecutions + metrics.executionStatusCount.numberOfFailedExecutions;
+        this.successfulExecutions = metrics.executionStatusCount.numberOfSuccessfulExecutions;
+        this.failedExecutions = metrics.executionStatusCount.numberOfFailedExecutions;
+      }
+    }
+  }
+
+  private insertToMetricsTable(
+    table: MetricsTableObject[],
+    metricsData: CpuMetric | MemoryMetric | ExecutionTimeMetric,
+    metricLabel: string
+  ) {
+    table.push({
+      metric: metricLabel,
+      min: metricsData?.minimum,
+      avg: metricsData?.average,
+      max: metricsData?.maximum,
+      unit: metricsData?.unit,
+    });
+  }
+
+  private loadValidationsData(partner: PartnerEnum) {
+    this.validationsTable = [];
+    const validations = this.metrics.get(partner).validationStatus?.validatorToolToIsValid;
+    if (validations) {
+      this.validationsColumns = [
+        'validatorTool',
+        'mostRecentVersion',
+        'isValid',
+        'mostRecentErrorMessage', // shows the error message (if exists) when isValid is false
+        'successfulValidationVersions',
+        'failedValidationVersions',
+        'numberOfRuns',
+        'passingRate',
+      ];
+      for (const [validatorTool, isValid] of Object.entries(validations)) {
+        this.validationsTable.push({
+          validatorTool: validatorTool,
+          mostRecentVersion: isValid.mostRecentVersion,
+          isValid: isValid.mostRecentIsValid,
+          successfulValidationVersions: isValid.successfulValidationVersions,
+          mostRecentErrorMessage: isValid?.mostRecentErrorMessage,
```

Comment:
I don't think you need the question mark. Or if you do need it, if `isValid` can be be null/undefined, then you need it on all lines between 185 and 191.